x-pack/plugins/observability_solution/apm/public/components/routing/templates/apm_main_template/index.tsx

```diff
@@ -77,8 +78,18 @@ export function ApmMainTemplate({
     entityCentricExperience,
     true
   );
-  const { isEntityCentricExperienceViewEnabled, serviceInventoryViewLocalStorageSetting } =
-    useEntityManagerEnablementContext();
+
+  const {
+    isEntityCentricExperienceViewEnabled,
+    serviceInventoryViewLocalStorageSetting,
+    setServiceInventoryViewLocalStorageSetting,
+  } = useEntityManagerEnablementContext();
+
+  useEffect(() => {
```

Comment:
Sure! Here’s a more polished version of your message:

---

@cauemarcondes, I've implemented a temporary fix to address this [comment](https://github.com/elastic/kibana/pull/193997#issuecomment-2374122587). It was necessary to apply this change after the component mounts, as the context wasn’t rendered yet, leading to runtime errors.

I also noticed that @kpatticha created an [issue](https://github.com/elastic/kibana/issues/194114) related to the cleanup of the services inventory view context. Once that task is completed, we can remove this temporary fix.